### DISCUSSION TITLE
Implementing Movable Words with WaveSurfer.js

### DISCUSSION THREAD

**Discussion by farhadjaman** [katspaugh/wavesurfer.js#3877] [General]
I want to add a feature with wavesurfer.js in my react app where:

- Words are displayed above the waveform at specific time points.
- These words can be clicked and dragged to adjust their timing.
- The new timing of the words is reflected in the component's state.

![image](https://github.com/user-attachments/assets/9852e769-de7e-4e50-b04d-094813b251fd)

**Comment by farhadjaman** [comment]
@katspaugh any idea please?

**Reply by katspaugh** [reply]
Sounds similar to https://wavesurfer.xyz/examples/?regions.js, perhaps you could customize the CSS of the regions to be above the waveform. See https://wavesurfer.xyz/examples/?styling.js for custom styling.
